demo

```diff
@@ -1,8 +1,8 @@
 #!/usr/bin/env bash
 set -e -u -o pipefail
 
-declare -r SCRIPT_PATH=$(readlink -f "$0")
-declare -r SCRIPT_DIR=$(cd $(dirname "$SCRIPT_PATH") && pwd)
+declare -r SCRIPT_PATH=$(perl -e "print readlink '$0'") 2>/dev/null
```

Comment:
why `perl` ( IMHO, unnecessary dependency) `bash` should be sufficient ? if you don't want to use `readlink` consider 

```
SCRIPT_DIR=$(cd -P $(dirname $0) && pwd)
```

Reply:
👍